sqlite-migrations/src/main/kotlin/com/squareup/sqlite/migrations/DatabaseFilesCollector.kt

```diff
@@ -1,20 +1,9 @@
 package com.squareup.sqlite.migrations
 
 import java.io.File
-import java.util.ArrayDeque
 
-object DatabaseFilesCollector {
-  inline fun forDatabaseFiles(sourceFolders: Iterable<File>, block: (File) -> Unit) {
-    val folders = ArrayDeque<File>().apply { sourceFolders.forEach(this::addLast) }
-    while (folders.isNotEmpty()) {
-      val folder = folders.removeFirst()
-      folder.listFiles()?.forEach { file ->
-        if (file.name.endsWith(".db")) {
-          block(file)
-        } else if (file.isDirectory) {
-          folders.addLast(file)
-        }
-      }
-    }
-  }
-}
+inline fun forDatabaseFiles(sourceFolders: Iterable<File>, callback: (dbFile: File) -> Unit) = sourceFolders
+  .asSequence()
+  .flatMap(File::walk)
+  .filter { entry -> entry.isFile && entry.name.endsWith(".db") }
+  .forEach(callback)
```

Comment:
This is possibly outside the scope of this PR, but perhaps we could have something like this, as it doesn't pollute the global namespace (and allows further chaining if we ever need it).

```suggestion
fun Iterable<File>.databaseFiles() = sourceFolders
  .asSequence()
  .flatMap(File::walk)
  .filter { entry -> entry.isFile && entry.name.endsWith(".db") }
```

Reply:
I like that!